build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -99,6 +99,9 @@ service Execution {
   // * `CANCELLED`: The operation was cancelled by the client. This status is
   //   only possible if the server implements the Operations API CancelOperation
   //   method, and it was called for the current execution.
+  // * `NOT_FOUND`: Due to a transient condition, an unknown operation name,
```

Comment:
I don't think that this error condition also applies to Execute(). That call doesn't take an operation name, right? It's specific to WaitExecution().

Reply:
Transiency could well occur during Execute, in the same timeframe that an Execute request might end, and a WaitExecution would be made. The errors that WaitExecution could see are also summarized here, and I made that explicit with the add below.

Reply:
> Transiency could well occur during Execute, in the same timeframe that an Execute request might end, and a WaitExecution would be made.

This part I don't really understand. All I'm saying is, can Execute() fail with `NOT_FOUND` because an invalid operation name was specified? My understand is that this cannot occur, because it is never called with an operation name.

My recommendation would be to leave the description of Execute() as is, but to replace this sentence that you added to WaitExecution:

> This method will report errors consistent with `Execute`.

With something along the lines of:

> In addition to the cases describe for `Execute`, the `WaitExecution` method may fail as follows:
>
> * `NOT_FOUND`: Description goes here.

Reply:
It's a subtle point, but I think specifying this with WaitExecution alone will probably not prevent Execute interpretation on clients from needing to handle it as a response to either method. The sentiment is that *if* a WaitExecution *would* return a NOT_FOUND at some point due to operation disappearance, then a currently active Execute call should also return that.

Putting it into the description for WaitExecution will suffice, I'll update.

Reply:
> The sentiment is that if a WaitExecution would return a NOT_FOUND at some point due to operation disappearance, then a currently active Execute call should also return that.

WaitExecution should basically do the following things:

1. Look up an operation by operation name.
2. Potentially attach/associate the current gRPC call to that operation, depending on what the implementation looks like.
3. Report events/mutations that occur against the operation.

I think you're stating above that NOT\_FOUND should be returned if an operation were to disappear during (3). But I don't think that's desirable. There are gRPC status codes that are far more relevant to situations like these, and we should use those instead. (CANCELLED? ABORTED? Pick one.)

In my opinion NOT\_FOUND should only be returned if step (1) fails. As step (1) is specific to WaitExecution and not Execute, there is thus no point in documenting NOT\_FOUND as a valid status code for Execute.

Reply:
Alright then. Moved to above WaitExecution in an updated commit already.